gremlin-javascript/src/main/javascript/gremlin-javascript/lib/structure/io/graph-serializer.js

```diff
@@ -50,27 +50,29 @@ class GraphSONWriter {
       }
       s.writer = this;
       // Insert custom serializers first
-      this._serializers.unshift(s);
+      th
+      is._serializers.unshift(s);
     });
   }
 
   adaptObject(value) {
     let s;
+    let o = value;
     if (Array.isArray(value)) {
-      return value.map(item => this.adaptObject(item));
+      o = value.map(item => this.adaptObject(item));
```

Comment:
I think that by serializing child items first in case of an `Array`, it causes to make two iterations of the `Array`: one here and the other one inside `ListSerializer`.

I think that, in order to maintain compat with GraphSON2 and still use `ListSerializer` for GraphSON3, we should evaluate the `if (Array.isArray(value))` condition below:

```javascript
// After the for loop
if (s) {
  s.serialize(o);
}

if (Array.isArray(value)) {
  return value.map(item => this.adaptObject(item));
}
```